src/codecs/resize/options.tsx

```diff
@@ -0,0 +1,128 @@
+import { h, Component } from 'preact';
+import linkState from 'linkstate';
+import { bind, inputFieldValueAsNumber } from '../../lib/util';
+import { ResizeOptions } from './resize';
+
+interface Props {
+  options: ResizeOptions;
+  aspect: number;
+  onChange(newOptions: ResizeOptions): void;
+}
+
+interface State {
+  maintainAspect: boolean;
+}
+
+export default class QuantizerOptions extends Component<Props, State> {
+  state: State = {
+    maintainAspect: true,
+  };
+
+  form: HTMLFormElement = undefined as any as HTMLFormElement;
+
+  reportOptions() {
+    const width = this.form.width as HTMLInputElement;
+    const height = this.form.height as HTMLInputElement;
+
+    if (!width.checkValidity() || !height.checkValidity()) return;
+
+    const options: ResizeOptions = {
+      width: inputFieldValueAsNumber(width),
+      height: inputFieldValueAsNumber(height),
+      method: this.form.resizeMethod.value,
+      fitMethod: this.form.fitMethod.value,
+    };
+    this.props.onChange(options);
+  }
+
+  @bind
+  onChange(event: Event) {
+    this.reportOptions();
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (!prevState.maintainAspect && this.state.maintainAspect) {
+      this.form.height.value = Math.round(Number(this.form.width.value) / this.props.aspect);
```

Comment:
Just wondering: Anything we should do to make sure `aspect > 0`?

Reply:
I'd expect an aspect of 0 to cause a decode error much sooner I think